pkg/solver/controller.go

```diff
@@ -448,8 +448,12 @@ func (controller *SolverController) addDeal(ctx context.Context, deal data.Deal)
 
 	controller.log.Info("add deal", deal)
 
+	//creates deal container and sets state to agreed
+	dealContainer := data.GetDealContainer(deal)
+	dealContainer.State = data.GetAgreementStateIndex("DealAgreed")
```

Comment:
[FYI] we'll need to change this to `DealCreated` in when the new contracts are out